test/test_torch.py

```diff
@@ -2975,6 +2975,15 @@ def test_is_deterministic_deprecated_warning(self, device):
             with self.assertWarnsOnceRegex(UserWarning, "torch.is_deterministic is deprecated"):
                 torch.is_deterministic()
 
+    # Validates that mathematical constants are defined properly, as required by
+    # the Python Array API (https://data-apis.org/array-api/latest/API_specification/constants.html)
+    @onlyCPU
+    def test_constants(self, device):
+        self.assertEqual(torch.e, math.e, atol=0, rtol=0)
+        self.assertEqual(torch.pi, math.pi, atol=0, rtol=0)
+        self.assertEqual(torch.nan, math.nan, equal_nan=True)
+        self.assertEqual(torch.inf, math.inf)
```

Comment:
One thing that this discards is the fact that the `torch` constants *must* by Python floats. As written, the test will also pass if the constants are 0d tensors:

https://github.com/pytorch/pytorch/blob/9e79a8a54f3e06ed613a0194e640c4622412dcfd/torch/testing/_internal/common_utils.py#L1284-L1286

So we need something like `isinstance(torch.e, float)` for every constant. 
```suggestion
        self.assertIsInstance(torch.e, float)
        self.assertEqual(torch.e, math.e, atol=0, rtol=0)
        
        self.assertIsInstance(torch.pi, float)
        self.assertEqual(torch.pi, math.pi, atol=0, rtol=0)
        
        self.assertIsInstance(torch.nan, float)
        self.assertEqual(torch.nan, math.nan, equal_nan=True)
        
        self.assertIsInstance(torch.inf, float)
        self.assertEqual(torch.inf, math.inf)
```